### PR TITLE
pubsub/doc: add a packagedoc section to each provider about Receive semantics

### LINKED ISSUE #1554: pubsub: add packagedoc to each driver about the semantics of send/receive
See https://github.com/google/go-cloud/issues/1368#issuecomment-472951496 for context.

### PR DESCRIPTION
**Pull request by vangent**
Fixes #1554.

### CODE REVIEW

File: pubsub/azuresb/azuresb.go [hunk 1]
```diff
@@ -26,6 +26,16 @@
 // see URLOpener.
 // See https://godoc.org/gocloud.dev#hdr-URLs for background information.
 //
+// Message Delivery Semantics
+//
+// Azure ServiceBus supports at-least-once semantics in the default Peek-Lock
+// mode; applications must call Message.Ack/Nack after processing a message, or
+// it will be redelivered. However, it also supports a Receive-Delete mode,
+// which essentially auto-acks a message when it is delivered, resulting in
+// at-most-once semantics. See SubscriberOptions.AckFuncForReceiveAndDelete.
```
Comment (jba) [inline pubsub/azuresb/azuresb.go]: Instead of just "See SubscriberOptions....", say something like "Use ... to choose between the two."

File: pubsub/awssnssqs/awssnssqs.go [hunk 2]
```diff
@@ -26,6 +26,13 @@
 // see URLOpener.
 // See https://godoc.org/gocloud.dev#hdr-URLs for background information.
 //
+// Message Delivery Semantics
+//
+// AWS SNS and SQS combine to support at-least-once semantics; applications must
+// call Message.Ack/Nack after processing a message, or it will be redelivered.
```
Comment (jba) [inline pubsub/awssnssqs/awssnssqs.go]: Well, if you call `Nack`, it _will_ be redelivered. So maybe reword.

Comment (vangent) [reply pubsub/awssnssqs/awssnssqs.go]: Right, done. I added the reference to Nack later. I removed it again, I don't think it's necessary to mention Nack here.

File: pubsub/azuresb/azuresb.go (hunk 1 above)
Comment (vangent) [reply pubsub/azuresb/azuresb.go]: Done.

**Comment by vangent** [comment]
Gentle ping @eliben ?
